From 6619bc3c2b7baa9edae3e9dc2b4ec03a676f580b Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 15 Sep 2017 11:13:17 +0800
Subject: [PATCH 2554/5242] MLK-16285-3 usb: cdns3: core: move INIT_WORK
 before cdns3_register_extcon

commit  75f6c641253535c71d170ccc8b7ed26fc25e09ee from
https://source.codeaurora.org/external/imx/linux-imx.git

At the extcon notifier, it will queue a work item, so we need to
make sure the work is initialized before it is used.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Acked-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/cdns3/core.c b/drivers/usb/cdns3/core.c
index e526018..ddc9fff 100644
--- a/drivers/usb/cdns3/core.c
+++ b/drivers/usb/cdns3/core.c
@@ -534,6 +534,7 @@ static int cdns3_probe(struct platform_device *pdev)
 	if (ret)
 		goto err1;
 
+	INIT_WORK(&cdns->role_switch_wq, cdns3_role_switch);
 	ret = cdns3_register_extcon(cdns);
 	if (ret)
 		goto err2;
@@ -557,7 +558,6 @@ static int cdns3_probe(struct platform_device *pdev)
 	if (ret)
 		goto err4;
 
-	INIT_WORK(&cdns->role_switch_wq, cdns3_role_switch);
 	dev_dbg(dev, "Cadence USB3 core: probe succeed\n");
 
 	return 0;
-- 
1.7.9.5

